src/keymgmt.c

```diff
@@ -1184,6 +1198,282 @@ const OSSL_DISPATCH p11prov_ec_keymgmt_functions[] = {
+{
+    P11PROV_OBJ *key = (P11PROV_OBJ *)keydata;
+    P11PROV_CTX *ctx = p11prov_obj_get_prov_ctx(key);
+
+    P11PROV_debug("ed export %p", keydata);
+
+    if (key == NULL) {
+        return RET_OSSL_ERR;
+    }
+
+    if (p11prov_ctx_allow_export(ctx) & DISALLOW_EXPORT_PUBLIC) {
+        return RET_OSSL_ERR;
+    }
+
+    /* this will return the public EC_POINT */
+    if ((selection & ~(PUBLIC_PARAMS)) == 0) {
+        return p11prov_obj_export_public_ec_key(key, cb_fn, cb_arg);
+    }
+
+    return RET_OSSL_ERR;
+}
+
+static const OSSL_PARAM *p11prov_ed_import_types(int selection)
+{
+    static const OSSL_PARAM p11prov_ed_imp_key_types[] = {
+        OSSL_PARAM_octet_string(OSSL_PKEY_PARAM_PRIV_KEY, NULL, 0),
+        OSSL_PARAM_octet_string(OSSL_PKEY_PARAM_PUB_KEY, NULL, 0),
+        OSSL_PARAM_END,
+    };
+    P11PROV_debug("ed import types");
+    if (selection & OSSL_KEYMGMT_SELECT_KEYPAIR) {
+        return p11prov_ed_imp_key_types;
+    }
+    return NULL;
+}
+
+static const OSSL_PARAM *p11prov_ed_export_types(int selection)
+{
+    static const OSSL_PARAM p11prov_ed_exp_key_types[] = {
+        OSSL_PARAM_octet_string(OSSL_PKEY_PARAM_PUB_KEY, NULL, 0),
+        OSSL_PARAM_END,
+    };
+    P11PROV_debug("ed export types");
+    if (selection == OSSL_KEYMGMT_SELECT_PUBLIC_KEY) {
+        return p11prov_ed_exp_key_types;
+    }
+    return NULL;
+}
+
+static const char *p11prov_ed25519_query_operation_name(int operation_id)
+{
+    if (operation_id == OSSL_OP_SIGNATURE) {
+        return P11PROV_NAME_ED25519;
+    }
+    return NULL;
+}
+
+static const char *p11prov_ed448_query_operation_name(int operation_id)
+{
+    if (operation_id == OSSL_OP_SIGNATURE) {
+        return P11PROV_NAME_ED448;
+    }
+    return NULL;
+}
+
+static int p11prov_ed_get_params(void *keydata, OSSL_PARAM params[])
+{
+    P11PROV_OBJ *key = (P11PROV_OBJ *)keydata;
+    OSSL_PARAM *p;
+    CK_ULONG group_size;
+    int ret;
+
+    P11PROV_debug("ed get params %p", keydata);
+
+    if (key == NULL) {
+        return RET_OSSL_ERR;
+    }
+
+    group_size = p11prov_obj_get_key_bit_size(key);
+    if (group_size == CK_UNAVAILABLE_INFORMATION) {
+        return RET_OSSL_ERR;
+    }
+
+    p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_BITS);
+    if (p) {
+        ret = OSSL_PARAM_set_int(p, group_size);
+        if (ret != RET_OSSL_OK) {
+            return ret;
+        }
+    }
+    p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_SECURITY_BITS);
+    if (p) {
+        int secbits;
+        if (group_size == 456) {
+            secbits = 224;
+        } else if (group_size == 256) {
+            secbits = 128;
+        } else {
+            return RET_OSSL_ERR;
+        }
+        ret = OSSL_PARAM_set_int(p, secbits);
+        if (ret != RET_OSSL_OK) {
+            return ret;
+        }
+    }
+    p = OSSL_PARAM_locate(params, OSSL_PKEY_PARAM_MAX_SIZE);
+    if (p) {
+        int sigsize;
+        if (group_size == 456) {
+            sigsize = 114;
```

Comment:
Would it make sense to reuse the constants `ED448_SIG_SIZE` and `ED25519_SIG_SIZE` here too?

Reply:
Oh yeah,
as I was developing I sprinkled these macros a bit everywhere, and then missed backtracking and checking all uses for consistency.
I am going to revisit the commits, pull all the macros in a single header and use them consistently.